prisma-fmt/src/lib.rs

```diff
@@ -2,6 +2,23 @@ mod actions;
 mod lint;
 mod native;
 mod preview;
+mod text_document_completion;
+
+/// The API is modelled on an LSP [completion
+/// request](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-16.md#textDocument_completion).
+/// Input and output are both JSON, the request being a `CompletionParams` object and the response
+/// being a `CompletionList` object.
+pub fn text_document_completion(schema: &str, params: String) -> String {
+    let params = if let Ok(params) = serde_json::from_str::<lsp_types::CompletionParams>(&params) {
+        params
+    } else {
+        return serde_json::to_string(&text_document_completion::empty_completion_list()).unwrap();
+    };
+
+    let completion_list = text_document_completion::completion(schema, params);
+
+    serde_json::to_string(&completion_list).unwrap()
+}
```

Comment:
This is the new, additional API I guess?

Reply:
yes exactly